src/DynamoCoreWpf/Extensions/ViewLoadedParams.cs

```diff
@@ -15,47 +19,67 @@ namespace Dynamo.Wpf.Extensions
     public class ViewLoadedParams
     {
         // TBD MAGN-7366
-        //
-        // Implementation notes:
-        // 
-        // This should be designed primarily to support the separation of the Package Manager from Core
-        // and minimize exposing unnecessary innards.
-        //
-        // It is expected that this class will be extended in the future, so it should stay as minimal as possible.
-        //
-        // Here's a start on the implementation
-        //
-
-        //private readonly DynamoView view;
-
-        //internal ViewLoadedParams(DynamoView view)
-        //{
-        //    this.view = view;
-        //}
-
-        ///// <summary>
-        ///// Add a menu item for workspace context click
-        ///// </summary>
-        ///// <param name="item">The item to insert</param>
-        ///// <param name="options">Options object to determine in which cases the MenuItem should be visible</param>
-        //public void AddWorkspaceContextClickMenuItem(MenuItem item, WorkspaceContextClickOptions options)
-        //{
-            
-        //}
 
-        //public void AddShortcutBarItem(ShortcutBarItem item)
-        //{
-        //    // add an item to Dynamo's shortcut bar
-        //}
+        private readonly DynamoView dynamoView;
+        private readonly DynamoViewModel dynamoViewModel;
+        private readonly Menu dynamoMenu;
 
-        //public void AddMenuBar(MenuItem item)
-        //{
-            
-        //}
+        private List<Tuple<MenuBarType, MenuItem>> addedMenuItems = new List<Tuple<MenuBarType, MenuItem>>();
+
+        public IEnumerable<IWorkspaceViewModel> WorkspaceViewModels
+        {
+            get
+            {
+                return dynamoViewModel.Workspaces;
+            }
+        }
 
-        //public void AddKeyBinding(KeyBinding binding)
-        //{
+        public ViewLoadedParams(DynamoView dynamoV, DynamoViewModel dynamoVM)
+        {
+            dynamoView = dynamoV;
+            dynamoViewModel = dynamoVM;
+            dynamoMenu = dynamoView.titleBar.ChildOfType<Menu>();
+
+            Disposable.Create(ClearMenuItems);
```

Comment:
Not quite.  I realize this will be more involved, so let's not consider it for now.  Let's remove this line, `ClearMenuItems`, and all of the other code associated with disposing the `MenuItems`.